worker/upgradedatabase/worker.go

```diff
@@ -280,9 +281,31 @@ func (w *upgradeDB) watchUpgrade() {
 	if err := w.upgradeInfo.Refresh(); err != nil {
 		w.logger.Errorf("unable to refresh upgrade info: %v", err)
 		w.setFailStatus()
-		return
+		return err
 	}
 
+	// To be here, this node previously returned false for isPrimary
+	// Sometimes our primary changes, or is reported as false when called too
+	// early. In the case that a node state changes whilst watching,
+	// esclate an error which will result in the worker being restarted
+	stateChanged := make(chan struct{}, 1)
```

Comment:
I don't think these need to be 1-buffered.

Reply:
It was originally when I wrote the basic idea, but now we're closing it, you're correct it doesn't need to be